src/main/java/Duke.java

```diff
@@ -1,10 +1,292 @@
+import java.util.Scanner;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    // defining some constants
+    public static final String INDENT = "     ";
+    public static final String TICK = "\u2713";
+    public static final String CROSS = "\u2717";
+    public static final String dirPath = "./data";
+    public static final String dataFilePath = "./data/duke.txt";
+
+    // init array of Task objects
+    public static ArrayList<Task> allTasks = new ArrayList<>();
+
+    // initialise task file data, print welcome message
+    public static void initProgram() {
+        File fDir = new File (dirPath);
+        File fTxt = new File (dataFilePath);
+
+        try {
+            readFileContents(dataFilePath);
+        } catch (FileNotFoundException e) {
+            if (!fDir.exists()) {
+                fDir.mkdir();
+            }
+            if (!fTxt.exists()) {
+                try {
+                    fTxt.createNewFile();
+                } catch (IOException ioException) {
+                    System.exit(1);
+                }
+            }
+        }
+
+        System.out.println(INDENT + "Hello! I'm Duke"
+                           + System.lineSeparator()
+                           + INDENT + "What can I do for you?");
+    }
+
+    // parse user input in REPL loop
+    public static void parseUserInput() {
+        Scanner line = new Scanner(System.in);
+        String input = line.nextLine();
+
+        String userKeyWord;
+
+        while(!input.toLowerCase().equals("bye")) {
+            String[] inputArr = input.split(" ");
+
+            userKeyWord = inputArr[0].toLowerCase();
+
+            switch (userKeyWord) {
+            case("todo"):
```

Comment:
Having no break statements in cases can lead to execution flowing into default block and introduce bugs.